backend/dataall/core/environment/api/resolvers.py

```diff
@@ -46,300 +24,154 @@ def get_trust_account(context: Context, source, **kwargs):
 
 
 def remove_group(context: Context, source, environmentUri=None, groupUri=None):
-    with context.engine.scoped_session() as session:
-        environment = EnvironmentService.remove_group(
-            session=session,
-            uri=environmentUri,
-            group=groupUri,
-        )
+    environment = EnvironmentService.remove_group(
+        uri=environmentUri,
+        group=groupUri,
+    )
 
     StackService.deploy_stack(targetUri=environment.environmentUri)
 
     return environment
 
 
 def remove_consumption_role(context: Context, source, environmentUri=None, consumptionRoleUri=None):
-    with context.engine.scoped_session() as session:
-        status = EnvironmentService.remove_consumption_role(
-            session=session,
-            uri=consumptionRoleUri,
-            env_uri=environmentUri,
-        )
+    status = EnvironmentService.remove_consumption_role(
+        uri=consumptionRoleUri,
+        env_uri=environmentUri,
+    )
 
     return status
 
 
 def update_consumption_role(context: Context, source, environmentUri=None, consumptionRoleUri=None, input={}):
-    with context.engine.scoped_session() as session:
-        consumption_role = EnvironmentService.update_consumption_role(
-            session=session,
-            uri=consumptionRoleUri,
-            env_uri=environmentUri,
-            input=input,
-        )
+    consumption_role = EnvironmentService.update_consumption_role(
+        uri=consumptionRoleUri,
+        env_uri=environmentUri,
+        input=input,
+    )
     return consumption_role
 
 
 def list_environment_invited_groups(context: Context, source, environmentUri=None, filter=None):
-    if filter is None:
-        filter = {}
-    with context.engine.scoped_session() as session:
-        return EnvironmentService.paginated_environment_invited_groups(
-            session=session,
-            uri=environmentUri,
-            data=filter,
-        )
+    return EnvironmentService.paginated_environment_invited_groups(
+        uri=environmentUri,
+        data=filter,
+    )
 
 
 def list_environment_groups(context: Context, source, environmentUri=None, filter=None):
-    if filter is None:
-        filter = {}
-    with context.engine.scoped_session() as session:
-        return EnvironmentService.paginated_user_environment_groups(
-            session=session,
-            uri=environmentUri,
-            data=filter,
-        )
+    return EnvironmentService.paginated_user_environment_groups(
+        uri=environmentUri,
+        data=filter,
+    )
 
 
 def list_all_environment_groups(context: Context, source, environmentUri=None, filter=None):
-    if filter is None:
-        filter = {}
-    with context.engine.scoped_session() as session:
-        return EnvironmentService.paginated_all_environment_groups(
-            session=session,
-            uri=environmentUri,
-            data=filter,
-        )
+    return EnvironmentService.paginated_all_environment_groups(
```

Comment:
For everywhere that we remove 
```
    if filter is None:
         filter = {}
```
Should we not set the filter default value = `{}`? 

Otherwise in the method called in EnvironmentService (such as `paginated_all_environment_groups()`) - would the method not fail with an Attribute Error `"'NoneType' object has no attribute 'get'"`

This may not be an issue for frontend when we always pass a default filter but what if we called API programmatically?

Reply:
UPDATE: After further review some of the methods in EnvironmentService do have the same logic passed down of:
```
if data:
     data = {}
```

The list of resolver methods that I presume would still need the above logic in their EnvironmentService function are:
- paginated_all_environment_consumption_roles
- paginated_user_environment_consumption_roles
- paginated_all_environment_groups
- paginated_user_environment_groups
- paginated_environment_invited_groups

Reply:
I checked every occurrence. Now in service there are checks\
` data = data if data is not None else {}`

Reply:
done